yunwoo/programmers-0/컨트롤-제트.js

```diff
@@ -0,0 +1,19 @@
+function solution(s) {
+  const sum = s.split(" ");
+
+  while (sum.includes("Z")) {
+    const zIdx = sum.findIndex((el) => el === "Z");
+    sum.splice(zIdx - 1, 2);
+    console.log(sum);
+  }
+
+  return sum.reduce((acc, cur) => acc + parseInt(cur), 0);
+}
```

Comment:
```suggestion
function solution(s) {
  const sum = s.split(" ");

  while (true) {
    const zIdx = sum.indexOf("Z");
      
    if (zIdx === -1) break;
    sum.splice(zIdx - 1, 2);
  }

  return sum.reduce((acc, cur) => acc + parseInt(cur), 0);
}
```
- 풀이 잘봤습니다 👍 그런데 while 조건문에서 includes를 하고 내부에서 또 findIndex를 하기 보다 while문 조건식을 그냥 true로 주고 내부에서 indexOf 메서드 하나만을 이용해서 -1이면 반복문을 종료하게끔 하면 연산 횟수를 더 줄여볼 수 있을 것 같습니다 ~!
- 아래 풀이는 스택을 이용한 시간복잡도 O(N) 풀이입니다.
```js
function solution(s) {
  const arr = s.split(" ");
  const stack = [];
  let result = 0;

  arr.forEach((el) => {
    if (el === "Z") {
      result -= stack.pop();
    } else {
      result += +el;
      stack.push(+el);
    }
  });

  return result;
}
```

Reply:
- 헉 `includes`를 사용할 필요가 없군요..! 감사합니다 참고하겠습니다!!
- 스택이 가능하다니..! 한개의 빈배열과 빈 변수를 이용하는거군요!!! 생각도 못했습니다.. 참고 풀이 감사합니다!!!🙇‍♂️